### ISSUE TITLE
soundwire sdw:2:25d:1308:0: Probe not complete, timed out

### ISSUE THREAD

**Issue by kcgthb** [thesofproject/linux#2986]
HI!

I'm observing a regression on a Dell Precision 5750 running Debian 5.10.0-7-amd64. 

Sound (output and mic) used to work just fine, but a recent update must have broken something, as no audio devices are detected anymore. Alsa config has not changed, firmware files are still in place (Debian package `firmware-sof-signed` 1.6.1-2), but suddenly, the device initialization fails to happen properly.

Here are the kernel logs:
```
[...]
[    9.200170] usbcore: registered new interface driver btusb
[    9.201268] Bluetooth: hci0: Firmware revision 0.0 build 121 week 7 2021
[    9.272816] iwlwifi 0000:00:14.3: base HW address: 98:af:65:cb:44:54
[    9.285627] thermal thermal_zone9: failed to read out thermal zone (-61)
[    9.606850] dell_laptop: Using i8042 filter function for receiving events
[    9.656050] sof-audio-pci 0000:00:1f.3: DSP detected with PCI class/subclass/prog-if info 0x040100
[    9.656070] sof-audio-pci 0000:00:1f.3: enabling device (0000 -> 0002)
[    9.656198] sof-audio-pci 0000:00:1f.3: DSP detected with PCI class/subclass/prog-if 0x040100
[    9.656330] sof-audio-pci 0000:00:1f.3: bound 0000:00:02.0 (ops i915_audio_component_bind_ops [i915])
[    9.663052] sof-audio-pci 0000:00:1f.3: use msi interrupt mode
[   10.004817] sof-audio-pci 0000:00:1f.3: hda codecs found, mask 4
[   10.012891] sof-audio-pci 0000:00:1f.3: firmware: direct-loading firmware intel/sof/sof-cml.ri
[   10.149982] sof-audio-pci 0000:00:1f.3: Firmware info: version 1:6:1-53680
[   10.149985] sof-audio-pci 0000:00:1f.3: Firmware: ABI 3:17:0 Kernel ABI 3:17:0
[   12.179894] soundwire sdw:2:25d:1308:0: Probe not complete, timed out
[   12.179897] soundwire sdw-master-2: Update Slave status failed:-110
[   12.179909] soundwire sdw:1:25d:1308:0: Probe not complete, timed out
[   12.179910] soundwire sdw-master-1: Update Slave status failed:-110
[   12.179916] soundwire sdw:0:25d:711:0: Probe not complete, timed out
[   12.179918] soundwire sdw-master-0: Update Slave status failed:-110
[   12.216315] soundwire sdw:3:25d:715:0: Probe not complete, timed out
[   12.216318] soundwire sdw-master-3: Update Slave status failed:-110
[   13.204624] ucsi_acpi USBC000:00: unknown error 0
[   13.204627] ucsi_acpi USBC000:00: UCSI_GET_PDOS failed (-5)
[   13.505796] intel_rapl_common: Found RAPL domain package
[   13.505798] intel_rapl_common: Found RAPL domain core
[   13.505799] intel_rapl_common: Found RAPL domain uncore
[   13.505800] intel_rapl_common: Found RAPL domain dram
[   13.505801] intel_rapl_common: Found RAPL domain psys
[   13.841323] iwlwifi 0000:00:14.3 wlp0s20f3: renamed from wlan0
[   13.859954] mei_hdcp 0000:00:16.0-b638ab7e-94e2-4ea2-a552-d1c54b627f04: bound 0000:00:02.0 (ops i915_hdcp_component_ops [i915])
[   13.927774] sof-audio-pci 0000:00:1f.3: firmware: direct-loading firmware intel/sof-tplg/sof-cml-rt711-rt1308-rt715.tplg
[   13.927784] sof-audio-pci 0000:00:1f.3: Topology: ABI 3:17:0 Kernel ABI 3:17:0
[   13.943007] sof-audio-pci 0000:00:1f.3: ASoC: Parent card not yet available, widget card binding deferred
[   13.966568] input: sof-soundwire Headset Jack as /devices/pci0000:00/0000:00:1f.3/sof_sdw/sound/card0/input27
[   13.966721] input: sof-soundwire HDMI/DP,pcm=5 as /devices/pci0000:00/0000:00:1f.3/sof_sdw/sound/card0/input28
[   13.966855] input: sof-soundwire HDMI/DP,pcm=6 as /devices/pci0000:00/0000:00:1f.3/sof_sdw/sound/card0/input29
[   13.967001] input: sof-soundwire HDMI/DP,pcm=7 as /devices/pci0000:00/0000:00:1f.3/sof_sdw/sound/card0/input30
[   16.882617] bbswitch: version 0.8
[   16.882621] bbswitch: Found integrated VGA device 0000:00:02.0: \_SB_.PCI0.GFX0
[   16.882628] bbswitch: Found discrete VGA device 0000:01:00.0: \_SB_.PCI0.PEG0.PEGP
[   16.882645] ACPI Warning: \_SB.PCI0.PEG0.PEGP._DSM: Argument #4 type mismatch - Found [Buffer], ACPI requires [Package] (20200925/nsarguments-61)
[   16.882718] bbswitch: detected an Optimus _DSM function
[   16.882728] bbswitch: Succesfully loaded. Discrete card 0000:01:00.0 is on
[   16.883087] [drm] [nvidia-drm] [GPU ID 0x00000100] Unloading driver
[   16.937447] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[   16.937449] Bluetooth: BNEP filters: protocol multicast
[   16.937451] Bluetooth: BNEP socket layer initialized
[   16.937685] nvidia-modeset: Unloading
[   16.985428] NET: Registered protocol family 38
[   16.986540] nvidia-nvlink: Unregistered the Nvlink Core, major device number 245
[   17.034020] bbswitch: disabling discrete graphics
[   17.053614] pci 0000:01:00.0: refused to change power state from D0 to D3hot
[   17.651917] xhci_hcd 0000:01:00.2: can't change power state from D3cold to D0 (config space inaccessible)
[   17.651940] xhci_hcd 0000:01:00.2: can't change power state from D3cold to D0 (config space inaccessible)
[   17.651943] xhci_hcd 0000:01:00.2: Controller not ready at resume -19
[   17.651944] xhci_hcd 0000:01:00.2: PCI post-resume error -19!
[   17.651945] xhci_hcd 0000:01:00.2: HC died; cleaning up
[   19.133761] rt711 sdw:0:25d:711:0: runtime PM trying to activate child device sdw:0:25d:711:0 but parent (sdw-master-0) is not active
[   19.135289] rt1308 sdw:1:25d:1308:0: DPN_PortCtrl register write failed for port 1
[   19.135291] soundwire sdw-master-1: Program transport params failed: -61
[   19.135292] soundwire sdw-master-1: Program params failed: -61
[   19.135294]  SDW1-Playback: ASoC: error at snd_soc_link_prepare on SDW1-Playback: -61
[   19.135295]  SDW1-Playback: ASoC: backend prepare failed -61
[   19.135466] sdw_deprepare_stream: subdevice #0-Playback: inconsistent state state 1
[   19.136373] rt1308 sdw:2:25d:1308:0: runtime PM trying to activate child device sdw:2:25d:1308:0 but parent (sdw-master-2) is not active
[   19.136725] rt1308 sdw:1:25d:1308:0: DPN_PortCtrl register write failed for port 1
[   19.136726] soundwire sdw-master-1: Program transport params failed: -61
[   19.136727] soundwire sdw-master-1: Program params failed: -61
[   19.136728]  SDW1-Playback: ASoC: error at snd_soc_link_prepare on SDW1-Playback: -61
[   19.136728]  SDW1-Playback: ASoC: backend prepare failed -61
[   19.136900] sdw_deprepare_stream: subdevice #0-Playback: inconsistent state state 1
[   19.138088] rt1308 sdw:1:25d:1308:0: DPN_PortCtrl register write failed for port 1
[   19.138089] soundwire sdw-master-1: Program transport params failed: -61
[   19.138090] soundwire sdw-master-1: Program params failed: -61
[   19.138090]  SDW1-Playback: ASoC: error at snd_soc_link_prepare on SDW1-Playback: -61
[   19.138091]  SDW1-Playback: ASoC: backend prepare failed -61
[   19.138257] sdw_deprepare_stream: subdevice #0-Playback: inconsistent state state 1
[   19.139445] rt1308 sdw:1:25d:1308:0: DPN_PortCtrl register write failed for port 1
[   19.139446] soundwire sdw-master-1: Program transport params failed: -61
[   19.139447] soundwire sdw-master-1: Program params failed: -61
[   19.139447]  SDW1-Playback: ASoC: error at snd_soc_link_prepare on SDW1-Playback: -61
[   19.139448]  SDW1-Playback: ASoC: backend prepare failed -61
[   19.139603] sdw_deprepare_stream: subdevice #0-Playback: inconsistent state state 1
[   19.140803] rt1308 sdw:1:25d:1308:0: DPN_PortCtrl register write failed for port 1
[   19.140805] soundwire sdw-master-1: Program transport params failed: -61
[   19.140805] soundwire sdw-master-1: Program params failed: -61
[   19.140806]  SDW1-Playback: ASoC: error at snd_soc_link_prepare on SDW1-Playback: -61
[   19.140807]  SDW1-Playback: ASoC: backend prepare failed -61
[   19.140975] sdw_deprepare_stream: subdevice #0-Playback: inconsistent state state 1
[   19.142294] rt1308 sdw:1:25d:1308:0: DPN_PortCtrl register write failed for port 1
[   19.142295] soundwire sdw-master-1: Program transport params failed: -61
[   19.142295] soundwire sdw-master-1: Program params failed: -61
[   19.142296]  SDW1-Playback: ASoC: error at snd_soc_link_prepare on SDW1-Playback: -61
[   19.142297]  SDW1-Playback: ASoC: backend prepare failed -61
[   19.142473] sdw_deprepare_stream: subdevice #0-Playback: inconsistent state state 1
[   19.143614] rt1308 sdw:1:25d:1308:0: DPN_PortCtrl register write failed for port 1
[   19.143615] soundwire sdw-master-1: Program transport params failed: -61
[   19.143616] soundwire sdw-master-1: Program params failed: -61
[   19.143617]  SDW1-Playback: ASoC: error at snd_soc_link_prepare on SDW1-Playback: -61
[   19.143617]  SDW1-Playback: ASoC: backend prepare failed -61
[   19.143787] sdw_deprepare_stream: subdevice #0-Playback: inconsistent state state 1
[   19.145008] rt1308 sdw:1:25d:1308:0: DPN_PortCtrl register write failed for port 1
[   19.145010] soundwire sdw-master-1: Program transport params failed: -61
[   19.145011] soundwire sdw-master-1: Program params failed: -61
[   19.145011]  SDW1-Playback: ASoC: error at snd_soc_link_prepare on SDW1-Playback: -61
[   19.145012]  SDW1-Playback: ASoC: backend prepare failed -61
[   19.145178] sdw_deprepare_stream: subdevice #0-Playback: inconsistent state state 1
[   19.146315] rt1308 sdw:1:25d:1308:0: DPN_PortCtrl register write failed for port 1
[   19.146316] soundwire sdw-master-1: Program transport params failed: -61
[   19.146317] soundwire sdw-master-1: Program params failed: -61
[   19.146318]  SDW1-Playback: ASoC: error at snd_soc_link_prepare on SDW1-Playback: -61
[   19.146318]  SDW1-Playback: ASoC: backend prepare failed -61
[   19.146488] sdw_deprepare_stream: subdevice #0-Playback: inconsistent state state 1
[   19.147692] rt1308 sdw:1:25d:1308:0: DPN_PortCtrl register write failed for port 1
[   19.147693] soundwire sdw-master-1: Program transport params failed: -61
[   19.147694] soundwire sdw-master-1: Program params failed: -61
[   19.147695]  SDW1-Playback: ASoC: error at snd_soc_link_prepare on SDW1-Playback: -61
[   19.147695]  SDW1-Playback: ASoC: backend prepare failed -61
[   19.147853] sdw_deprepare_stream: subdevice #0-Playback: inconsistent state state 1
[   19.153714] rt1308 sdw:1:25d:1308:0: runtime PM trying to activate child device sdw:1:25d:1308:0 but parent (sdw-master-1) is not active
[   19.400390] Failed to set private value: ffffffc3 <= 752039 40960
[   19.401179] rt715 sdw:3:25d:715:0: DPN_PortCtrl register write failed for port 4
[   19.401181] soundwire sdw-master-3: Program transport params failed: -61
[   19.401182] soundwire sdw-master-3: Program params failed: -61
[   19.401183]  SDW3-Capture: ASoC: error at snd_soc_link_prepare on SDW3-Capture: -61
[   19.401183]  SDW3-Capture: ASoC: backend prepare failed -61
[   19.401321] sdw_deprepare_stream: subdevice #0-Capture: inconsistent state state 1
[   19.401533] Failed to set private value: ffffffc3 <= 752039 40960
[   19.402283] rt715 sdw:3:25d:715:0: DPN_PortCtrl register write failed for port 4
[   19.402284] soundwire sdw-master-3: Program transport params failed: -61
[   19.402285] soundwire sdw-master-3: Program params failed: -61
[   19.402286]  SDW3-Capture: ASoC: error at snd_soc_link_prepare on SDW3-Capture: -61
[   19.402286]  SDW3-Capture: ASoC: backend prepare failed -61
[   19.402426] sdw_deprepare_stream: subdevice #0-Capture: inconsistent state state 1
[   19.402636] Failed to set private value: ffffffc3 <= 752039 40960
[   19.403408] rt715 sdw:3:25d:715:0: DPN_PortCtrl register write failed for port 4
[   19.403409] soundwire sdw-master-3: Program transport params failed: -61
[   19.403410] soundwire sdw-master-3: Program params failed: -61
[   19.403411]  SDW3-Capture: ASoC: error at snd_soc_link_prepare on SDW3-Capture: -61
[   19.403412]  SDW3-Capture: ASoC: backend prepare failed -61
[   19.403550] sdw_deprepare_stream: subdevice #0-Capture: inconsistent state state 1
[   19.403762] Failed to set private value: ffffffc3 <= 752039 40960
[   19.404664] rt715 sdw:3:25d:715:0: DPN_PortCtrl register write failed for port 4
[   19.404667] soundwire sdw-master-3: Program transport params failed: -61
[   19.404667] soundwire sdw-master-3: Program params failed: -61
[   19.404669]  SDW3-Capture: ASoC: error at snd_soc_link_prepare on SDW3-Capture: -61
[   19.404669]  SDW3-Capture: ASoC: backend prepare failed -61
[   19.404841] sdw_deprepare_stream: subdevice #0-Capture: inconsistent state state 1
[   19.405075] Failed to set private value: ffffffc3 <= 752039 40960
[   19.405857] rt715 sdw:3:25d:715:0: DPN_PortCtrl register write failed for port 4
[   19.405859] soundwire sdw-master-3: Program transport params failed: -61
[   19.405860] soundwire sdw-master-3: Program params failed: -61
[   19.405861]  SDW3-Capture: ASoC: error at snd_soc_link_prepare on SDW3-Capture: -61
[   19.405862]  SDW3-Capture: ASoC: backend prepare failed -61
[   19.406022] sdw_deprepare_stream: subdevice #0-Capture: inconsistent state state 1
[   19.406366] Failed to set private value: ffffffc3 <= 752039 40960
[   19.407136] rt715 sdw:3:25d:715:0: DPN_PortCtrl register write failed for port 4
[   19.407137] soundwire sdw-master-3: Program transport params failed: -61
[   19.407138] soundwire sdw-master-3: Program params failed: -61
[   19.407139]  SDW3-Capture: ASoC: error at snd_soc_link_prepare on SDW3-Capture: -61
[   19.407140]  SDW3-Capture: ASoC: backend prepare failed -61
[   19.407281] sdw_deprepare_stream: subdevice #0-Capture: inconsistent state state 1
[   19.407490] Failed to set private value: ffffffc3 <= 752039 40960
[   19.408240] rt715 sdw:3:25d:715:0: DPN_PortCtrl register write failed for port 4
[   19.408241] soundwire sdw-master-3: Program transport params failed: -61
[   19.408242] soundwire sdw-master-3: Program params failed: -61
[   19.408243]  SDW3-Capture: ASoC: error at snd_soc_link_prepare on SDW3-Capture: -61
[   19.408243]  SDW3-Capture: ASoC: backend prepare failed -61
[   19.408381] sdw_deprepare_stream: subdevice #0-Capture: inconsistent state state 1
[   19.408574] Failed to set private value: ffffffc3 <= 752039 40960
[   19.409345] rt715 sdw:3:25d:715:0: DPN_PortCtrl register write failed for port 4
[   19.409346] soundwire sdw-master-3: Program transport params failed: -61
[   19.409347] soundwire sdw-master-3: Program params failed: -61
[   19.409348]  SDW3-Capture: ASoC: error at snd_soc_link_prepare on SDW3-Capture: -61
[   19.409348]  SDW3-Capture: ASoC: backend prepare failed -61
[   19.409487] sdw_deprepare_stream: subdevice #0-Capture: inconsistent state state 1
[   19.409699] Failed to set private value: ffffffc3 <= 752039 40960
[   19.410448] rt715 sdw:3:25d:715:0: DPN_PortCtrl register write failed for port 4
[   19.410449] soundwire sdw-master-3: Program transport params failed: -61
[   19.410450] soundwire sdw-master-3: Program params failed: -61
[   19.410451]  SDW3-Capture: ASoC: error at snd_soc_link_prepare on SDW3-Capture: -61
[   19.410451]  SDW3-Capture: ASoC: backend prepare failed -61
[   19.410588] sdw_deprepare_stream: subdevice #0-Capture: inconsistent state state 1
[   19.410803] Failed to set private value: ffffffc3 <= 752039 40960
[   19.411554] rt715 sdw:3:25d:715:0: DPN_PortCtrl register write failed for port 4
[   19.411554] soundwire sdw-master-3: Program transport params failed: -61
[   19.411555] soundwire sdw-master-3: Program params failed: -61
[   19.411556]  SDW3-Capture: ASoC: error at snd_soc_link_prepare on SDW3-Capture: -61
[   19.411556]  SDW3-Capture: ASoC: backend prepare failed -61
[   19.411685] sdw_deprepare_stream: subdevice #0-Capture: inconsistent state state 1
[   19.419663] rt715 sdw:3:25d:715:0: runtime PM trying to activate child device sdw:3:25d:715:0 but parent (sdw-master-3) is not active
[   21.624407] Bluetooth: RFCOMM TTY layer initialized
[   21.624411] Bluetooth: RFCOMM socket layer initialized
```

The first error seems to be `soundwire sdw:2:25d:1308:0: Probe not complete, timed out`

Any idea what may cause this? The hardware is functional and works normally when booting Windows, so a hardware issue can be ruled out.

Thanks!

**Comment by plbossart** [comment]
@kcgthb thanks for reporting this.

It's likely a kernel issue, would you know what the previous config was? They should have kept the older kernel in theory during the update so you might be able to boot into the old version?

Would you mind reporting what 'sudo dmidecode' lists for the SKU number (that's how we track Dell devices). I suspect you have one of 098F or 0990 since it looks like you have the two-amplifier version.

The other thing would be to share your kernel config (or what Debian uses). The failure you have is rather spectacular in that NONE of the 4 devices present on SoundWire links are probed. 

listing all the modules loaded, e.g. by providing a link to the results of the 'alsa-info' script would be useful.

It could be that the drivers are not compiled in and the probe fails because there's just nothing to deal with the hardware.

**Comment by kcgthb** [comment]
@plbossart Thanks for your reply!

> would you know what the previous config was?

I've only used stock Debian kernels, so yes, I have the config that's provided in the package.
The diff between the last 2 versions is like this:
```
 diff -u /boot/config-5.10.0-6-amd64 /boot/config-5.10.0-7-amd64
--- /boot/config-5.10.0-6-amd64 2021-04-09 11:17:58.000000000 -0700
+++ /boot/config-5.10.0-7-amd64 2021-05-28 01:31:38.000000000 -0700
@@ -1,6 +1,6 @@
 #
 # Automatically generated file; DO NOT EDIT.
-# Linux/x86 5.10.28 Kernel Configuration
+# Linux/x86 5.10.40 Kernel Configuration
 #
 CONFIG_CC_VERSION_TEXT="gcc-10 (Debian 10.2.1-6) 10.2.1 20210110"
 CONFIG_CC_IS_GCC=y
@@ -23,7 +23,7 @@
 # CONFIG_COMPILE_TEST is not set
 CONFIG_LOCALVERSION=""
 # CONFIG_LOCALVERSION_AUTO is not set
-CONFIG_BUILD_SALT="5.10.0-6-amd64"
+CONFIG_BUILD_SALT="5.10.0-7-amd64"
 CONFIG_HAVE_KERNEL_GZIP=y
 CONFIG_HAVE_KERNEL_BZIP2=y
 CONFIG_HAVE_KERNEL_LZMA=y
@@ -2464,8 +2464,8 @@
 CONFIG_ALTERA_STAPL=m
 CONFIG_INTEL_MEI=m
 CONFIG_INTEL_MEI_ME=m
-# CONFIG_INTEL_MEI_TXE is not set
-# CONFIG_INTEL_MEI_HDCP is not set
+CONFIG_INTEL_MEI_TXE=m
+CONFIG_INTEL_MEI_HDCP=m
 CONFIG_VMWARE_VMCI=m
 # CONFIG_GENWQE is not set
 # CONFIG_ECHO is not set
@@ -3939,7 +3939,7 @@
 CONFIG_RMI4_F12=y
 CONFIG_RMI4_F30=y
 CONFIG_RMI4_F34=y
-# CONFIG_RMI4_F3A is not set
+CONFIG_RMI4_F3A=y
 # CONFIG_RMI4_F54 is not set
 CONFIG_RMI4_F55=y
 
@@ -6303,7 +6303,7 @@
 # CONFIG_SND_SOC_IMG is not set
 CONFIG_SND_SOC_INTEL_SST_TOPLEVEL=y
 CONFIG_SND_SOC_INTEL_SST=m
-# CONFIG_SND_SOC_INTEL_CATPT is not set
+CONFIG_SND_SOC_INTEL_CATPT=m
 CONFIG_SND_SST_ATOM_HIFI2_PLATFORM=m
 # CONFIG_SND_SST_ATOM_HIFI2_PLATFORM_PCI is not set
 CONFIG_SND_SST_ATOM_HIFI2_PLATFORM_ACPI=m
@@ -6323,6 +6323,10 @@
 CONFIG_SND_SOC_ACPI_INTEL_MATCH=m
 CONFIG_SND_SOC_INTEL_MACH=y
 CONFIG_SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES=y
+CONFIG_SND_SOC_INTEL_HASWELL_MACH=m
+CONFIG_SND_SOC_INTEL_BDW_RT5650_MACH=m
+CONFIG_SND_SOC_INTEL_BDW_RT5677_MACH=m
+CONFIG_SND_SOC_INTEL_BROADWELL_MACH=m
 CONFIG_SND_SOC_INTEL_BYTCR_RT5640_MACH=m
 CONFIG_SND_SOC_INTEL_BYTCR_RT5651_MACH=m
 CONFIG_SND_SOC_INTEL_CHT_BSW_RT5672_MACH=m
@@ -6506,6 +6510,8 @@
 CONFIG_SND_SOC_RT5651=m
 CONFIG_SND_SOC_RT5663=m
 CONFIG_SND_SOC_RT5670=m
+CONFIG_SND_SOC_RT5677=m
+CONFIG_SND_SOC_RT5677_SPI=m
 CONFIG_SND_SOC_RT5682=m
 CONFIG_SND_SOC_RT5682_I2C=m
 CONFIG_SND_SOC_RT5682_SDW=m
@@ -8065,8 +8071,6 @@
 CONFIG_AD7923=m
 CONFIG_AD7949=m
 CONFIG_AD799X=m
-CONFIG_AD9467=m
-CONFIG_ADI_AXI_ADC=m
 CONFIG_AXP20X_ADC=m
 CONFIG_AXP288_ADC=m
 CONFIG_CC10001_ADC=m
 ```

> They should have kept the older kernel in theory during the update so you might be able to boot into the old version?

Yes. Now that's the very very weird thing: booting on an older kernel seems to produce the same result now, with the same device initialization failure... I haven't been very careful in noting the versions between reboots and I haven't checked that audio was working after every boot lately (I mostly used Bluetooth headphones over the last few weeks) so I'm not exactly sure when things stopped working. :\

> Would you mind reporting what 'sudo dmidecode' lists for the SKU number (that's how we track Dell devices). I suspect you have one of 098F or 0990 since it looks like you have the two-amplifier version.

Good guess! :)
```
# dmidecode -s system-sku-number
0990
```

> The other thing would be to share your kernel config (or what Debian uses). The failure you have is rather spectacular in that NONE of the 4 devices present on SoundWire links are probed.

Ok, I just now realize that the log I initially provided was with `snd_hda_intel` blacklisted, which may not have been very helpful.
Here's the full `dmesg` of a boot with `snd_hda_intel`: https://hastebin.com/raw/yerukasedu

It actually starts almost the same:
```
Jun  8 14:22:30 mauna kernel: [    9.855874] sof-audio-pci 0000:00:1f.3: Firmware info: version 1:6:1-53680
Jun  8 14:22:30 mauna kernel: [    9.855876] sof-audio-pci 0000:00:1f.3: Firmware: ABI 3:17:0 Kernel ABI 3:17:0
Jun  8 14:22:30 mauna kernel: [   11.891199] soundwire sdw:2:25d:1308:0: Probe not complete, timed out
Jun  8 14:22:30 mauna kernel: [   11.891203] soundwire sdw-master-2: Update Slave status failed:-110
Jun  8 14:22:30 mauna kernel: [   11.891214] soundwire sdw:0:25d:711:0: Probe not complete, timed out
Jun  8 14:22:30 mauna kernel: [   11.891216] soundwire sdw-master-0: Update Slave status failed:-110
Jun  8 14:22:30 mauna kernel: [   11.891225] soundwire sdw:3:25d:715:0: Probe not complete, timed out
Jun  8 14:22:30 mauna kernel: [   11.891226] soundwire sdw-master-3: Update Slave status failed:-110
Jun  8 14:22:30 mauna kernel: [   11.891233] soundwire sdw:1:25d:1308:0: Probe not complete, timed out
Jun  8 14:22:30 mauna kernel: [   11.891234] soundwire sdw-master-1: Update Slave status failed:-110
```
but then it goes:
```
Jun  8 14:22:32 mauna kernel: [   21.391326] intel-sdw intel-sdw.3: IO transfer timed out, cmd 3 device 1 addr 3501 len 1
Jun  8 14:22:32 mauna kernel: [   21.391329] soundwire sdw-master-3: trf on Slave 1 failed:-110
Jun  8 14:22:34 mauna kernel: [   23.407497] intel-sdw intel-sdw.3: IO transfer timed out, cmd 3 device 1 addr 7307 len 1
Jun  8 14:22:34 mauna kernel: [   23.407509] soundwire sdw-master-3: trf on Slave 1 failed:-110
```

So I don't know if it makes much of a difference or not. In any case, the kernel config is here:
https://hastebin.com/raw/bojarucera

> listing all the modules loaded, e.g. by providing a link to the results of the 'alsa-info' script would be useful.

Sure, here it is: http://alsa-project.org/db/?f=8277785ef44e03b2b9afaab8af59ba1c96160351

> It could be that the drivers are not compiled in and the probe fails because there's just nothing to deal with the hardware.

I've read about the required `CONFIG_SND_SOC_INTEL_USER_FRIENDLY_LONG_NAMES` option for those machines, but apparently the option is set.

In any case, thank you!

**Comment by carnil** [comment]
For reference only, downstream kernel report for the issue: https://bugs.debian.org/989622

**Comment by plbossart** [comment]
Thanks @kcgthb I am still in the dark on what happens. It seems you have a card so the configuration seems ok.

We will need more traces to figure this out, can you please add this file 
[sof-dyndbg.conf.txt](https://github.com/thesofproject/linux/files/6624379/sof-dyndbg.conf.txt) as /etc/modprobe.d/sof-dyndbg.conf, reboot and attach the full dmesg log.

That would help us see more that the flow is. Thanks!

**Comment by kcgthb** [comment]
Absolutely, yes. Please find the full dmesg at https://hastebin.com/raw/ceguvibone (EDIT: sorry initial links contained incomplete logs).

Thank you very much for looking into this!

**Comment by kcgthb** [comment]
Ok, and after waiting for a while, it looks like the devices end up being initialized after all (hence the updated log in my previous comment):
```
$ aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: NVidia [HDA NVidia], device 3: HDMI 0 [HDMI 0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: NVidia [HDA NVidia], device 7: HDMI 1 [HDMI 1]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: NVidia [HDA NVidia], device 8: HDMI 2 [HDMI 2]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: NVidia [HDA NVidia], device 9: HDMI 3 [HDMI 3]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: NVidia [HDA NVidia], device 10: HDMI 4 [HDMI 4]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: NVidia [HDA NVidia], device 11: HDMI 5 [HDMI 5]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: NVidia [HDA NVidia], device 12: HDMI 6 [HDMI 6]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: sofsoundwire [sof-soundwire], device 0: Headphone (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: sofsoundwire [sof-soundwire], device 2: SDW1-speakers (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: sofsoundwire [sof-soundwire], device 5: HDMI1 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: sofsoundwire [sof-soundwire], device 6: HDMI2 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: sofsoundwire [sof-soundwire], device 7: HDMI3 (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
$ arecord -l
**** List of CAPTURE Hardware Devices ****
card 1: sofsoundwire [sof-soundwire], device 1: Headset mic (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: sofsoundwire [sof-soundwire], device 4: Microphones (*) []
  Subdevices: 1/1
  Subdevice #0: subdevice #0
  ```
  
  So I'm not sure where the `IO transfer timed out` errors are coming from, could it be a power management issue?

**Comment by kcgthb** [comment]
Well, I may have been a bit quick jumping to conclusion that the device was properly initialized. Although it seems to be detected, PulseAudio doesn't seem to be able to do much with it: See the output of a `systemctl --user start pulseaudio.service` at https://hastebin.com/raw/abazohoyam

```
soundwire sdw-master-1: Compute bus params failed: -22
soundwire sdw-master-1: Compute params failed: -22
  SDW1-Playback: ASoC: error at snd_soc_link_prepare on SDW1-Playback: -22
  SDW1-Playback: ASoC: backend prepare failed -22
[...]
Failed to load module "module-alsa-card" (argument: "device_id="1" name="pci-0000_00_1f.3-platform-sof_sdw" card_name="alsa_card.pci-0000_00_1f.3-platform-sof_sdw" namereg_fail=false tsched=yes fixed_latency_range=no ignore_dB=no deferred_volume=yes use_ucm=yes avoid_resampling=no card_properties="module-udev-detect.discovered=1""): initialization failed.
```

**Comment by plbossart** [comment]
Thanks @kcgthb this helps.

It could be that the probe timeout is too small.

The sequence starts well

````
[    9.106521] soundwire sdw-master-0: SDW Slave class_id 0, part_id 711, mfg_id 25d, unique_id 0, version 2
[    9.107469] soundwire sdw:0:25d:711:0: sdw_modify_slave_status: signaling completion for Slave 1
````

but then goes South
````
[   11.119174] soundwire sdw:1:25d:1308:0: Probe not complete, timed out
````

The timeout is 2s in drivers/soundwire/bus.h:

#define DEFAULT_PROBE_TIMEOUT       2000

This timeout was added to make sure that the modules are loaded and probed *BEFORE* we start calling into the codec driver. We need to have a number of variables that need to be initialized in the probe, otherwise the code goes boink.

Something on your platform causes the modules to be loaded with a delay.

if you have the ability to recompile your kernel, can you try to update this timeout to e.g. 10000 (10 seconds).

**Comment by kcgthb** [comment]
> if you have the ability to recompile your kernel, can you try to update this timeout to e.g. 10000 (10 seconds).

So, I did just that, and wow, spot on: the device now comes up right away at boot, the timeout messages are gone.
The full dmesg with `sof-dyndbg.conf` and a timeout of 10s is at https://hastebin.com/raw/hukopofazi

Thank you!

**Comment by plbossart** [comment]
Cool, thanks @kcgthb for testing.

Can I bother you to try https://github.com/thesofproject/linux/commit/23f091b1072b165b982c4fdadde618e500e13c9d.patch (keeping the added timemout) and resend the dmesg log.

I'd like to see if 10s are actually necessary and size the timing changes more accurately.

**Comment by kcgthb** [comment]
Sure! I actually got a compilation error with the patch, using the default Debian compiler: 
`gcc version 10.2.1 20210110 (Debian 10.2.1-6)`
```
In file included from ./include/linux/kernel.h:16,
                 from ./include/linux/list.h:9,
                 from ./include/linux/module.h:12,
                 from drivers/soundwire/bus_type.c:4:
drivers/soundwire/bus_type.c: In function ‘__sdw_register_driver’:
drivers/soundwire/bus_type.c:196:42: error: ‘name’ undeclared (first use in this function)
  196 |  pr_info("plb: registering driver %s\n", name);
      |                                          ^~~~
./include/linux/printk.h:373:34: note: in definition of macro ‘pr_info’
  373 |  printk(KERN_INFO pr_fmt(fmt), ##__VA_ARGS__)
      |                                  ^~~~~~~~~~~
drivers/soundwire/bus_type.c:196:42: note: each undeclared identifier is reported only once for each function it appears in
  196 |  pr_info("plb: registering driver %s\n", name);
      |                                          ^~~~
./include/linux/printk.h:373:34: note: in definition of macro ‘pr_info’
  373 |  printk(KERN_INFO pr_fmt(fmt), ##__VA_ARGS__)
      |     
```

**Comment by kcgthb** [comment]
I gave a try at the patch with: 
```
pr_info("plb: registering driver %s\n", drv->name);
``` 
instead of:
```
pr_info("plb: registering driver %s\n", name);
```
which compiled fine. 

But I'm afraid the next boot sent the kernel in some sort of infinite loop, with the kernel spewing logs at a rate that made the machine unusable (see the timestamps):
```
Jun  9 15:15:47 mauna kernel: [   18.991585] rt711 sdw:0:25d:711:0: plb: setting modalias sdw:m025Dp0711v02c00
Jun  9 15:15:47 mauna kernel: [   18.991585]
Jun  9 15:15:47 mauna kernel: [   18.991657] rt715 sdw:3:25d:715:0: plb: setting modalias sdw:m025Dp0715v02c00
Jun  9 15:15:47 mauna kernel: [   18.991657]
Jun  9 15:15:47 mauna kernel: [   18.991729] rt711 sdw:0:25d:711:0: plb: setting modalias sdw:m025Dp0711v02c00
Jun  9 15:15:47 mauna kernel: [   18.991729]
Jun  9 15:15:47 mauna kernel: [   18.991800] rt1308 sdw:1:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.991800]
Jun  9 15:15:47 mauna kernel: [   18.991872] rt711 sdw:0:25d:711:0: plb: setting modalias sdw:m025Dp0711v02c00
Jun  9 15:15:47 mauna kernel: [   18.991872]
Jun  9 15:15:47 mauna kernel: [   18.991951] rt1308 sdw:2:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.991951]
Jun  9 15:15:47 mauna kernel: [   18.992023] rt1308 sdw:1:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.992023]
Jun  9 15:15:47 mauna kernel: [   18.992094] rt1308 sdw:2:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.992094]
Jun  9 15:15:47 mauna kernel: [   18.992164] rt1308 sdw:2:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.992164]
Jun  9 15:15:47 mauna kernel: [   18.992236] rt1308 sdw:2:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.992236]
Jun  9 15:15:47 mauna kernel: [   18.992307] rt715 sdw:3:25d:715:0: plb: setting modalias sdw:m025Dp0715v02c00
Jun  9 15:15:47 mauna kernel: [   18.992307]
Jun  9 15:15:47 mauna kernel: [   18.992378] rt715 sdw:3:25d:715:0: plb: setting modalias sdw:m025Dp0715v02c00
Jun  9 15:15:47 mauna kernel: [   18.992378]
Jun  9 15:15:47 mauna kernel: [   18.992450] rt1308 sdw:1:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.992450]
Jun  9 15:15:47 mauna kernel: [   18.992608] rt715 sdw:3:25d:715:0: plb: setting modalias sdw:m025Dp0715v02c00
Jun  9 15:15:47 mauna kernel: [   18.992608]
Jun  9 15:15:47 mauna kernel: [   18.992681] rt1308 sdw:2:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.992681]
Jun  9 15:15:47 mauna kernel: [   18.992753] rt715 sdw:3:25d:715:0: plb: setting modalias sdw:m025Dp0715v02c00
Jun  9 15:15:47 mauna kernel: [   18.992753]
Jun  9 15:15:47 mauna kernel: [   18.992831] rt715 sdw:3:25d:715:0: plb: setting modalias sdw:m025Dp0715v02c00
Jun  9 15:15:47 mauna kernel: [   18.992831]
Jun  9 15:15:47 mauna kernel: [   18.992903] rt1308 sdw:1:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.992903]
Jun  9 15:15:47 mauna kernel: [   18.992980] rt711 sdw:0:25d:711:0: plb: setting modalias sdw:m025Dp0711v02c00
Jun  9 15:15:47 mauna kernel: [   18.992980]
Jun  9 15:15:47 mauna kernel: [   18.993051] rt711 sdw:0:25d:711:0: plb: setting modalias sdw:m025Dp0711v02c00
Jun  9 15:15:47 mauna kernel: [   18.993051]
Jun  9 15:15:47 mauna kernel: [   18.993122] rt1308 sdw:2:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.993122]
Jun  9 15:15:47 mauna kernel: [   18.993192] rt1308 sdw:1:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.993192]
Jun  9 15:15:47 mauna kernel: [   18.993265] rt711 sdw:0:25d:711:0: plb: setting modalias sdw:m025Dp0711v02c00
Jun  9 15:15:47 mauna kernel: [   18.993265]
Jun  9 15:15:47 mauna kernel: [   18.993337] rt1308 sdw:1:25d:1308:0: plb: setting modalias sdw:m025Dp1308v02c00
Jun  9 15:15:47 mauna kernel: [   18.993337]
Jun  9 15:15:47 mauna kernel: [   18.993408] rt711 sdw:0:25d:711:0: plb: setting modalias sdw:m025Dp0711v02c00
Jun  9 15:15:47 mauna kernel: [   18.993408]
Jun  9 15:15:47 mauna kernel: [   18.993480] rt715 sdw:3:25d:715:0: plb: setting modalias sdw:m025Dp0715v02c00
```

Not sure if that was caused by the `name` -> `drv->name` change, since the looping message is `setting modalias` while the `name` change was in the `registering driver` line. ;|

Thanks!

**Comment by plbossart** [comment]
Sorry @kcgthb, I forgot that modalias was super verbose

Please try with https://github.com/thesofproject/linux/commit/9674883e67f9699d777bbed0fda47b4d3da2bd14.patch
we should now see one message for each device.

**Comment by plbossart** [comment]
Sorry wrong version, this should be https://github.com/thesofproject/linux/commit/f13a75b8c218d377aac63a769badaee4506b1a26.patch

**Comment by plbossart** [comment]
Results from the Intel tests:

````
[    4.745136] kernel: soundwire sdw:0:025d:0701:00: plb: setting modalias sdw:m025Dp0701v01c00
[    4.752088] kernel: plb: registering driver rt700
[    4.752500] kernel: rt700 sdw:0:025d:0700:00: plb: probe complete
[    4.752913] kernel: rt700 sdw:1:025d:0700:00: plb: probe complete

[    4.752177] kernel: plb: registering driver rt1308
[    4.752427] kernel: rt1308 sdw:1:025d:1308:00:2: plb: probe complete
[    4.752668] kernel: rt1308 sdw:1:025d:1308:00:0: plb: probe complete

[    4.772712] kernel: plb: registering driver rt711
[    4.773001] kernel: rt711 sdw:0:025d:0711:00: plb: probe complete

-> 28ms for the probe.
````

2s timeout is plenty for this case.

**Comment by kcgthb** [comment]
Thanks for the updated patch! 

I was able to collect timing logs, and on my system, there are a couple probe timeouts that significantly increase the whole probing time:

```
[    8.642538] soundwire sdw:0:25d:711:0: plb: setting modalias sdw:m025Dp0711v02c00

[    8.645467] sof-audio-pci 0000:00:1f.3: hda codecs found, mask 4
[    8.648016] sof-audio-pci 0000:00:1f.3: firmware: direct-loading firmware intel/sof/sof-cml.ri

[    8.770598] sof-audio-pci 0000:00:1f.3: Firmware info: version 1:6:1-53680
[    8.770600] sof-audio-pci 0000:00:1f.3: Firmware: ABI 3:17:0 Kernel ABI 3:17:0

[   15.083666] plb: registering driver rt711
[   15.084212] plb: registering driver rt1308
[   15.121466] plb: registering driver rt715
[   15.143194] sof_sdw sof_sdw: snd_soc_register_card failed -517

[   18.963173] soundwire sdw:0:25d:711:0: Probe not complete, timed out
[   18.963177] soundwire sdw-master-0: Update Slave status failed:-110
[   18.963188] soundwire sdw:3:25d:715:0: Probe not complete, timed out
[   18.963189] soundwire sdw-master-3: Update Slave status failed:-110
[   18.963197] soundwire sdw:2:25d:1308:0: Probe not complete, timed out
[   18.963198] soundwire sdw-master-2: Update Slave status failed:-110
[   18.963206] soundwire sdw:1:25d:1308:0: Probe not complete, timed out
[   18.963208] soundwire sdw-master-1: Update Slave status failed:-110
[   18.964346] rt711 sdw:0:25d:711:0: plb: probe complete
[   18.964438] sof_sdw sof_sdw: snd_soc_register_card failed -517
[   18.964523] rt1308 sdw:1:25d:1308:0: plb: probe complete
[   18.964603] sof_sdw sof_sdw: snd_soc_register_card failed -517
[   18.964771] rt715 sdw:3:25d:715:0: plb: probe complete
[   18.964870] sof_sdw sof_sdw: snd_soc_register_card failed -517
[   18.965287] rt1308 sdw:2:25d:1308:0: plb: probe complete
```
The complete dmesg log is at https://hastebin.com/raw/ulidusocow
Please let me know if you'd like it with `sof-dyndbg.conf` as well.

Thanks!

**Comment by kcgthb** [comment]
Here's the full dmesg log with `dyndbg`: 
https://hastebin.com/raw/ajiwidobif

**Comment by plbossart** [comment]
@kcgthb that's very helpful, very puzzling results indeed.

so if we take a single device we have

````
[    8.642538] soundwire sdw:0:25d:711:0: plb: setting modalias sdw:m025Dp0711v02c00
[   15.083666] plb: registering driver rt711
[   18.964346] rt711 sdw:0:25d:711:0: plb: probe complete
````

That's more than 10s between the kernel asking udev to load a module and the driver probe completing.

My results show 28ms :-)

I don't know what could possibly cause this sort of delays on a device that is well supported by other distributions, e.g. Ubuntu never reported this sort of problems with their OEM image. 

I'll have to ask my esteemed colleagues @lgirdwood @lyakh @kv2019i @bardliao @RanderWang for advice here on what could cause this - either in the kernel configuration or userspace. 

the HDaudio driver seems to use up to 60s when doing a request_module("i915") but in this case there's no hardware initialization needed, just loading a module and executing the probe routine which does minimal memory allocation and initialization of flags.

**Comment by kcgthb** [comment]
> That's more than 10s between the kernel asking udev to load a module and the driver probe completing.

Yep, and in that particular case, the audio device didn't show up, the 10s timeout was not enough. :(

On the next reboot, the overall time was shorter, and the device was properly initialized. So it looks like the delay is not exactly identical across reboots.

Looking at the whole boot log picture, there seems to be a 3s gap here where nothing is logged:
```
[    8.929091] mauna kernel: intel-sdw intel-sdw.2: Slave status change
[    8.929101] mauna kernel: soundwire sdw:2:25d:1308:0: sdw_modify_slave_status: signaling completion for Slave 1
[    8.985841] mauna kernel: dell_laptop: Using i8042 filter function for receiving events
[   11.983898] mauna kernel: BTRFS info (device dm-0): devid 1 device path /dev/mapper/nvme0n1p5_crypt changed to /dev/dm-0 scanned by systemd-udevd (589)
[   12.041176] mauna kernel: BTRFS info (device dm-0): devid 1 device path /dev/dm-0 changed to /dev/mapper/nvme0n1p5_crypt scanned by systemd-udevd (589)
``` 

Not sure how relevant that could be, though.

**Comment by plbossart** [comment]
Results from @bardliao on the same device (https://github.com/thesofproject/linux/pull/2988#issuecomment-859225815)

````
[    3.634807] soundwire sdw:0:025d:0711:00: plb: setting modalias sdw:m025Dp0711v02c00
[    3.648600] plb: registering driver rt711
[    3.650132] rt711 sdw:0:025d:0711:00: plb: probe complete

[    3.649729] plb: registering driver rt1308
[    3.650447] rt1308 sdw:1:025d:1308:00: plb: probe complete
[    3.651112] rt1308 sdw:2:025d:1308:00: plb: probe complete


[    3.650737] plb: registering driver rt715
[    3.652306] rt715 sdw:3:025d:0715:00: plb: probe complete
````

Less than 18ms to load modules and probe all necessary drivers.

@kcgthb there's *something* in your setup that causes the delay, it's not a hardware or SOF issue.

**Comment by kcgthb** [comment]
Interesting! 
I wonder what the differences are that could explain these hugely different timings. 
Here's what I noted between @bardliao's setup and mine:

| | **@kcgthb**  | **@bardliao** |
| ---- | ---- | ---- |
| **SKU** | 0990 | 098F |
| **BIOS** | 1.7.2 03/04/2021 | 11.2.20 01/03/2020 |
| **distribution** | Debian 11.0 | Ubuntu 20.04 |
| **kernel** | 5.10.40 (+Debian patches) |  5.13.0-rc5+ (+patches?)  |

I also see that @bardliao's kernel logs contain plenty of messages from `sof-audio-pci-intel-cnl`, but mine don't: they're mostly from `sof-audio-pci`. Could that explain those differences too?

**Comment by plbossart** [comment]
@kcgthb we split sof-audio-pci in two in a recent kernel, I think 5.12, but there was no change in functionality, just moving the code around to solve compilation issues.

At this point I don't think there's a clear explanation, we'll probably have to test/bisect to root cause this.

I think we should first try a kernel update to see if this changes anything. We can provide you with a Debian package of our development kernel, or since you are obviously able to recompile your own kernel try with https://github.com/thesofproject/linux/commits/topic/sof-dev, using your existing debian kernel config (make olddefconfig to upgrade).

**Comment by plbossart** [comment]
@kcgthb here's our daily test kernel debian packages if it helps. The source commit is 57440285

[soundwire-test.tar.gz](https://github.com/thesofproject/linux/files/6640667/soundwire-test.tar.gz)

This is provided without any support or warranty, all standard disclaimers apply. Use at your own risk.

**Comment by kcgthb** [comment]
Thanks! I'll give that a try.

**Comment by kcgthb** [comment]
Unfortunately, that kernel produced `cryptsetup` errors right from the start, and wasn't able to mount my laptop's partitions, so I can't test it. 

I'll try patching my current 5.10 with `sof-dev`.

**Comment by kotarou3** [comment]
Just chiming in since I also found an audio-related kernel regression with 5.10.0-7-amd64, but nothing to do with sof: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=989778

It's probably unrelated, since https://github.com/thesofproject/linux/issues/2986#issuecomment-859658326 mentions there's no problem on v5.13.0-rc5+, while I also encountered my issue on that version too.  
But letting you know just in case

**Comment by plbossart** [comment]
> Unfortunately, that kernel produced `cryptsetup` errors right from the start, and wasn't able to mount my laptop's partitions, so I can't test it.
> 
> I'll try patching my current 5.10 with `sof-dev`.

Thanks @kcgthb 
Wondering if you have any sort of disk encryption possibly resulting in modules getting delayed while the authentication takes place.
I am not sure if you can just patch your system, we probably have thousands of additional patches on top of 5.10. You might as well try directly topic/sof-dev from our tree with the Debian config updated with 'make olddefconfig'. that's how I would do it.

**Comment by kcgthb** [comment]
@plbossart I do use encryption indeed, with LUKS, but AFAIK, once the whole partition is unlocked, there shouldn't be any additional authentication, nor added delay in loading modules.

Today, booting the same kernel as last week (Debian's 5.10.40 with your [timing patch](https://github.com/thesofproject/linux/commit/f13a75b8c218d377aac63a769badaee4506b1a26.patch)), I got the following:
```
#  dmesg | grep plb
[    8.111669] soundwire sdw:0:25d:711:0: plb: setting modalias sdw:m025Dp0711v02c00
[    8.346533] plb: registering driver rt711
[    8.346614] rt711 sdw:0:25d:711:0: plb: probe complete
[    8.349253] plb: registering driver rt1308
[    8.349301] rt1308 sdw:1:25d:1308:0: plb: probe complete
[    8.349347] rt1308 sdw:2:25d:1308:0: plb: probe complete
[    8.352679] plb: registering driver rt715
[    8.352757] rt715 sdw:3:25d:715:0: plb: probe complete
```
and no reported timeout whatsoever (see the full [`dmesg`](https://hastebin.com/raw/ruxugimune)).

The only thing I can think of is that I rebooted this laptop to its Windows 10 partition and usually hibernates from there. I'm wondering if the hibernated state could somehow leave the audio devices is a hung state that would generate those timeouts when they're initialized?

**Comment by plbossart** [comment]
@kcgthb sorry I am not following. Can you describe the sequence that works and the one that doesn't?

These routines that we traced are not tied to any sort of hardware state, they only deal with ACPI interfaces describing the hardware. But ACPI also has methods for power management that may get in the way. That's rather beyond my area of expertise :-)

**Comment by kcgthb** [comment]
> Can you describe the sequence that works and the one that doesn't?

That's exactly what I'm trying to figure :)

Last week, booting that same kernel produced "Probe not complete, timed out" messages. Today, it doesn't anymore. The only thing that happened in-between was booting Windows.

So yes, it's definitely puzzling. I'll try to reboot with the regular Debian kernels to see if I can reproduce the timeouts again.

**Comment by plbossart** [comment]
@kcgthb closing for now since there's no update or information pointing to an SOF issue.

**Comment by bphd** [comment]
Same but two years later, and no Windows thing at all
